### PR TITLE
Revert "docs: inline `document-features` usage, remove dep."

### PR DESCRIPTION
**Pull request by ErichDonGubler**
This reverts commit 3d5bec659b9cf19f1c64274de0d11808d771cc66, with an update to `document-features`, and preferring to keep new `feature` content. To be clear, the only difference I have observed is the addition of the `serde` feature.

**Connections**
_Link to the issues addressed by this PR, or dependent PRs in other repositories_

As stated in the commit message: (hopefully intelligently) reverts #4924.

**Description**
_Describe what problem this is solving, and how it's solved._

**Testing**
_Explain how this change is tested._

This is tested by checking docs.

**Checklist**

- ~~Run `cargo fmt`.~~
- ~~Run `cargo clippy`. If applicable, add:~~
  - ~~`--target wasm32-unknown-unknown`~~
  - ~~`--target wasm32-unknown-emscripten`~~
- ~~Run `cargo xtask test` to run tests.~~
- ~~Add change to `CHANGELOG.md`. See simple instructions inside file.~~

### CODE REVIEW

File: wgpu-core/src/lib.rs [hunk 1]
```diff
@@ -3,36 +3,8 @@
 //! into other language-specific user-friendly libraries.
 //!
 //! ## Feature flags
-// NOTE: feature docs. below should be kept in sync. with `Cargo.toml`!
+#![doc = document_features::document_features!()]
```
Comment (ErichDonGubler) [inline wgpu-core/src/lib.rs:6]: todo: Validate HTML output manually.

File: wgpu/src/lib.rs [hunk 2]
```diff
@@ -3,55 +3,7 @@
...
-//! - **`metal`** _(enabled by default)_ --- Enables the Metal backend on macOS & iOS.
-//! - **`webgpu`** _(enabled by default)_ --- Enables the WebGPU backend on Wasm. Disabled when targeting `emscripten`.
-//! - **`angle`** --- Enables the GLES backend via [ANGLE](https://github.com/google/angle) on macOS.
-//! - **`vulkan-portability`** --- Enables the Vulkan backend on macOS & iOS.
-//! - **`webgl`** --- Enables the GLES backend on Wasm.
-//!     - ⚠️ WIP: Currently will also enable GLES dependencies on any other targets.
-//!
-//! **Note:** In the documentation, if you see that an item depends on a backend,
-//! it means that the item is only available when that backend is enabled _and_ the backend
-//! is supported on the current platform.
```
Comment (ErichDonGubler) [inline wgpu/src/lib.rs:24]: todo: I don't see this paragraph in any TOML documents, ATM. Need to track this down in history.

Comment (cwfitzgerald) [reply wgpu/src/lib.rs:24]: This was me - I added it directly to the Readme. I didn't realize we were planning on going back.

Comment (ErichDonGubler) [reply wgpu/src/lib.rs:24]: I mean, we could! If everybody is fine with that, I can drop this PR. 😊 My intent here was to give the community back the nice thing that Firefox couldn't have for a while.

File: wgpu-core/src/lib.rs (hunk 1 above)
Comment (ErichDonGubler) [reply wgpu-core/src/lib.rs:6]: todo: Update commit message with context that was requested in Matrix: https://matrix.to/#/!FZyQrssSlHEZqrYcOb:matrix.org/$Wi893tpzWm45KyZfB7Jrcx7wa1lDWLmUDxz8w_ZX6o4?via=mozilla.org&via=matrix.org&via=envs.net

EDIT: Resolved with e6de854f0.

File: wgpu/src/lib.rs (hunk 2 above)
Comment (Wumpf) [reply wgpu/src/lib.rs:24]: oh please let's go back. Imho it's way better with `document-features`

**Review by cwfitzgerald — DISMISSED** [review]
This needs the changes in the readme looked at before merging - requesting changes for record keeping.

File: wgpu/src/lib.rs (hunk 2 above)
Comment (ErichDonGubler) [reply wgpu/src/lib.rs:24]: @cwfitzgerald: uh

I think you mean crate documentation instead? 🤔

File: wgpu-core/src/lib.rs (hunk 1 above)
Comment (ErichDonGubler) [reply wgpu-core/src/lib.rs:6]: Done!

File: wgpu/src/lib.rs (hunk 2 above)
Comment (ErichDonGubler) [reply wgpu/src/lib.rs:24]: I've reconciled differences between the current contents of `wgpu/Cargo.toml` and `lib.rs` with fb6289158 and 328bf5551. Differences of note:

* The `**Note:**` paragraph this conversation is attached to was missing from `Cargo.toml`.
* The `serde` feature's documentation was missing from `lib.rs`.
* The `webgpu` feature was in different positions between them.

…thus demonstrating that a single source of truth here is already going to be beneficial, despite the short window of time in which these two documents diverged. 😅

I also validated that `wgpu-core/Cargo.toml` and `wgpu-core/src/lib.rs` did not diverge meaningfully.

**Review by Wumpf — APPROVED** [review]

**Pull request merged**